core/math/a_star_grid_2d.cpp

```diff
@@ -734,6 +734,119 @@ TypedArray<Vector2i> AStarGrid2D::get_id_path(const Vector2i &p_from_id, const V
 	return path;
 }
 
+Vector2 AStarGrid2D::get_flow_direction(const Vector2i &p_id) const {
+	ERR_FAIL_COND_V_MSG(dirty, Vector2(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!is_in_boundsv(p_id), Vector2(), vformat("Can't get point's flow direction. Point %s out of bounds %s.", p_id, region));
+	return _get_point_unchecked(p_id)->flow_direction;
+}
+
+TypedArray<Vector2> AStarGrid2D::get_flow_directions(const Rect2i &p_region) const {
+	ERR_FAIL_COND_V_MSG(dirty, TypedArray<Vector2i>(), "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!region.encloses(p_region), TypedArray<Vector2i>(), vformat("Can't get point's flow direction. Region %s out of bounds %s.", p_region, region));
+
+	TypedArray<Vector2> result;
+	result.resize(p_region.size.x * p_region.size.y);
+
+	uint32_t index = 0U;
+	for (int y = p_region.get_position().y; y < p_region.get_end().y; y++) {
+		for (int x = p_region.get_position().x; x < p_region.get_end().x; x++) {
+			result[index++] = _get_point_unchecked(Vector2i(x, y))->flow_direction;
+		}
+	}
+	return result;
+}
+
+void AStarGrid2D::set_point_cost(const Vector2i &p_id, int p_cost) {
+	ERR_FAIL_COND_MSG(dirty, "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_MSG(!is_in_boundsv(p_id), vformat("Can't set point's cost. Point %s out of bounds %s.", p_id, region));
+	ERR_FAIL_COND_MSG(p_cost < 1 || p_cost > 254, "Can't set point's cost. Cost must be within (1-254) range.");
+	_get_point_unchecked(p_id)->cost = p_cost;
+}
+
+int AStarGrid2D::get_point_cost(const Vector2i &p_id) const {
+	ERR_FAIL_COND_V_MSG(dirty, 0, "Grid is not initialized. Call the update method.");
+	ERR_FAIL_COND_V_MSG(!is_in_boundsv(p_id), 0, vformat("Can't get point's cost. Point %s out of bounds %s.", p_id, region));
+	return _get_point_unchecked(p_id)->cost;
+}
+
+void AStarGrid2D::calculate_flow_field(const Vector2i &p_id, bool p_ignore_obstacles) {
```

Comment:
I think it would be good to have a version that could take a subset of the whole A star grid 
(`void AStarGrid2D::calculate_flow_field(const Vector2i &p_id, bool p_ignore_obstacles, Rect2i subregion)` ). I could definitely see a use case in games to have a huge AStarGrid for your level, but wanted to recalculate only subportions for it in some cases (like give me the flow field only 10 units around this unit). Maybe that could be in a follow up PR if the need arise. But the code would need to change just very little, only the indexes part will change and you'll need a distinction between width and pitch.

Reply:
Done, also removed ignore_obstacles since the algorithm is changed and not the whole grid is scanned for directions...